### DISCUSSION TITLE
Run prepare on an existing octoprint/ustreamer setup?

### DISCUSSION THREAD

**Discussion by matthieu-sofa** [paukstelis/octoprint_deploy#116] [Q&A] [answered]
Hi Paul,

Thanks a lot for octoprint_deploy, I watched your YT video and it seems to address my need!
Maybe 2 dummy questions:
1) I'm just wondering if there is a way to run the prepare script on an existing octoprint mono instance setup to not start form a blank octoprint config (i.e. having printer1 already configured with plugins, API key, etc)?

2) I would like to install a plugin like SpoolManager or Filamentmanager, so does the database of those plugins will be shared across instances? 

Thanks a lot.

Cheers from France

**Comment by paukstelis** [comment, marked as answer]
Hi Matt,

1. I started working on a method for detecting existing OctoPrint binaries and instances during the prepare stage. However, this proved to be fairly complex. Your best option is to make a back up of your existing instance, start with a fresh system installation while making an instance with the same name as your old instance, then restore the backup. This isn't perfect and you will need to adjust some settings afterwards, but it is the best solution I have for now.
2. By default the databases are not shared. It is possible to make them shared using symlinks. Decide which  instance will house the main database, then for your other instances, delete the existing database and make a symlink that points to the main database.

**Reply by matthieu-sofa** [reply]
Thanks a lot Paul
